### PR TITLE
Retry streams

### PR DESCRIPTION
**Pull request by grahamc**
##### Checklist

- [ ] Tested changes against a test repository
- [ ] Added or updated relevant documentation (leave unchecked if not applicable)
- [ ] (If this PR is for a release) Updated README to point to the new tag (leave unchecked if not applicable)

### CODE REVIEW

File: src/ids-host.ts [hunk 1]
```diff
@@ -17,7 +17,7 @@ const ALLOWED_SUFFIXES = [
 const DEFAULT_IDS_HOST = "https://install.determinate.systems";
 const LOOKUP = process.env["IDS_LOOKUP"] ?? DEFAULT_LOOKUP;
 
-const DEFAULT_TIMEOUT = 30_000; // 30 seconds in milliseconds
+const DEFAULT_TIMEOUT = 5; // 5ms
```
Comment (cole-h) [inline src/ids-host.ts:20]: oh?

Comment (grahamc) [reply src/ids-host.ts:20]: Forcing the error conditions to take place to make sure it is doing the right thing

Comment (cole-h) [reply src/ids-host.ts:20]: Ah, gotcha. I was not have expecting the comment to change in that case :sweat_smile:

**Review by colemickens — APPROVED** [review]

File: src/index.ts [hunk 2]
```diff
@@ -667,6 +698,56 @@ export abstract class DetSysAction {
...
+        });
+
+        writeStream.on("finish", () => {
+          if (!failed) {
+            resolve(stream);
+          }
+        });
+
+        stream.once("retry", (_count, _error, createRetryStream) => {
+          // Optional: check `failed' here in case you want to stop retrying
```
[the remark this answers is not in the document]
Comment (grahamc) [reply src/index.ts:737]: Hmmm... checking. Good shout.

[the remark this answers is not in the document]
Comment (cole-h) [reply src/index.ts:737]: Where do we configure this "retryCount"?

[the remark this answers is not in the document]
Comment (grahamc) [reply src/index.ts:737]: https://github.com/DeterminateSystems/detsys-ts/pull/63/files#diff-de32a1854fcbf2f024d1857854bd37e2c6d9e5848e071cdd62d65fa31befdffaL52-R57

**Review by lucperkins — APPROVED** [review]

**Pull request merged**